pkg/destroy/gcp/instance.go

```diff
@@ -35,15 +35,15 @@ func (o *ClusterUninstaller) getInstanceNameAndZone(instanceURL string) (string,
 }
 
 func (o *ClusterUninstaller) listInstances(ctx context.Context) ([]cloudResource, error) {
-	byName, err := o.listInstancesWithFilter(ctx, "items/*/instances(name,zone,status,machineType),nextPageToken",
+	byName, err := o.listInstancesWithFilter(ctx, "items/*/instances(name,zone,status,machineType,labels),nextPageToken",
 		func(item *compute.Instance) bool {
 			return o.isClusterResource(item.Name)
 		})
 	if err != nil {
 		return nil, err
 	}
 
-	byLabel, err := o.listInstancesWithFilter(ctx, "items/*/instances(name,zone,status,machineType),nextPageToken",
+	byLabel, err := o.listInstancesWithFilter(ctx, "items/*/instances(name,zone,status,machineType,labels),nextPageToken",
```

Comment:
If you are requesting the same fields in both calls, why not make it only one call and filter by:
```go
func(item *compute.Instance) bool {
    if o.isClusterResource(item) {
        return true
    }
    for key, value := range item.Labels {
        [...]
    }
    [...]
```